beacon-chain/blockchain/service.go

```diff
@@ -195,43 +189,26 @@ func (c *ChainService) updateHead(processedBlock <-chan *types.Block) {
 			}
 
 			// TODO(#674): Handle chain reorgs.
-			var newCState *types.CrystallizedState
-			if c.unfinalizedBlocks[h].cycleTransition {
-				newCState = blockcState
+			var newState *types.BeaconState
+			if c.unfinalizedBlocks[h].IsCycleTransition(block.SlotNumber()) {
+				newState = blockState
 			}
-			if err := c.beaconDB.UpdateChainHead(newHead, c.unfinalizedBlocks[h].activeState, newCState); err != nil {
+			if err := c.beaconDB.UpdateChainHead(newHead, newState); err != nil {
 				log.Errorf("Failed to update chain: %v", err)
 				continue
 			}
 			log.WithField("blockHash", fmt.Sprintf("0x%x", h)).Info("Chain head block and state updated")
 			// We fire events that notify listeners of a new block (or crystallized state in
 			// the case of a state transition). This is useful for the beacon node's gRPC
 			// server to stream these events to beacon clients.
-			if c.unfinalizedBlocks[h].cycleTransition {
-				c.canonicalCrystallizedStateFeed.Send(blockcState)
+			if c.unfinalizedBlocks[h].IsCycleTransition(block.SlotNumber()) {
+				c.canonicalStateFeed.Send(newState)
```

Comment:
Merging `active` and `crystallized` causes the same issue here. If I recall correctly, subscribers for this feed care about new committee assignments. If so, we may want to reconsider some of the logic here to make the intention more clear.

Reply:
you mean they are about knowing when the validator set changed?